src/App.tsx

```diff
@@ -1,39 +1,60 @@
-import React from 'react';
+import { Component } from 'react';
+
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
```

Comment:
initialize state inside constructor

Reply:
Had so many troubles with it. I couldn`t assign to timerId further, cuz it`s ReadOnly? also some type errors and eslint asks for usage as class property, not inside constructor. Is it really bertter option? In this video https://mate.academy/learn/react/class-components#/video/470 Misha says that it is fine to write as I did?